lightning/src/ln/channelmanager.rs

```diff
@@ -13048,6 +13057,17 @@ where
 			Some(hrn) => builder.sourced_from_human_readable_name(hrn),
 		};
 
+		let contacts = match contacts {
+			None => ContactSecrets::new(self.entropy_source.get_secure_random_bytes()),
+			Some(c) => c,
+		};
+		let builder = builder.contact_secrets(contacts.clone());
+		// Create a minimal offer for BLIP-42 contact exchange (just node_id, no description/paths)
+		// TODO: Create a better minimal offer with a single blinded path hop for privacy,
+		// while keeping the size small enough to fit in the onion packet.
+		let payer_offer = self.create_offer_builder()?.build()?;
```

Comment:
If we do keep always passing an offer instead of doing it conditionally, we should probably consider including (encrypted) data in the offer so that we can tie return payments back to the offer we paid (incl, eg, the BIP 353 we used to look up the name to begin with).